src/core.jl

```diff
@@ -278,15 +279,12 @@ drawwiththickness!(img::AbstractArray{T,2}, p::Point, color::T, in_bounds::Bool,
 drawwiththickness!(img::AbstractArray{T,2}, p::CartesianIndex{2}, color::T, in_bounds::Bool, thickness::Integer) where {T<:Colorant} = drawifinbounds!(img, Point(p), color, in_bounds, thickness)
 
 function drawwiththickness!(img::AbstractArray{T,2}, y0::Int, x0::Int, color::T, in_bounds::Bool, thickness::Int) where {T<:Colorant}
-    n = Int(round(thickness / 2))
-    evn = thickness % 2 == 1 ? 0 : 1
-    pixels = [i for i = -(n-evn):n]
+    m = ceil(Int, thickness/2) - 1
+    n = thickness % 2 == 0 ? m+1 : m
+    pixels = [i for i = -m:n]
 
-    for (x,y) in Combinatorics.combinations(pixels, 2)
+    for x in pixels, y in pixels
         draw!(img, y0+y, x0+x, color, in_bounds=in_bounds)
```

Comment:
Will it be possible when `thickness` is large enough, `in_bounds` should be `false`? Currently, if you pass `in_bounds=true`, it would always be `true`.

Reply:
What I'm worry about is when `in_bounds = true`, it will hit `OutOfBoundary` error for large
`(x, y)` in

```julia
for x in pixels, y in pixels
    draw!(img, y0+y, x0+x, color, in_bounds=in_bounds)
```

Reply:
🤔 I see, but perhaps a warning is more friendly than error? I'm not very sure of it.

This doesn't belong to this PR, we can rework `drawinbounds!` in the future if there's a need.

Reply:
>  let it draw objects even if it goes off the image.

We can define some boundary methods as types and pass it to `draw!`, just like what `ImageFiltering` does to `imfilter`.

I'm totally fine with the current design when ImageDraw isn't fully functional enough. But in the future, we need to bring this back to the table.